### PR TITLE
Bug 1173473 - Create sign-up form for special edition t-shirt.

### PR DESCRIPTION
**Pull request by kmaglione**
This form will only be live for a few weeks, and the code and database entries will be removed after the deadline passes. This doesn't touch any unrelated code to a significant degree, so it should be fairly safe to push out of band.

Given the short lifetime of this code, I haven't written any tests, but have tested manually. Let me know if you want view tests, but I'm pretty sure that they're not worth the effort.

### CODE REVIEW

File: apps/users/models.py [hunk 1]
```diff
@@ -38,6 +38,39 @@
...
+    user = models.ForeignKey('UserProfile', unique=True)
+
+    name = models.CharField(max_length=512)
+    address1 = models.CharField(max_length=512)
+    address2 = models.CharField(max_length=512, blank=True)
+    city = models.CharField(max_length=512)
+    state = models.CharField(max_length=512, blank=True)
+    zip = models.CharField(max_length=32, blank=True)
+    country = models.CharField(max_length=512)
+    telephone = models.CharField(max_length=512)
```
Comment (magopian) [inline apps/users/models.py]: Is this field mandatory?

File: apps/users/templates/users/t-shirt.html [hunk 2]
```diff
@@ -0,0 +1,137 @@
+{% extends "impala/base_shared.html" %}
+{% from 'includes/forms.html' import pretty_field, required %}
+
+{% block title %}{{ page_title(_('T-shirt Order')) }}{% endblock %}
+
+{% block bodyclass %}t-shirt-form{% endblock %}
+
+{% block content %}
+<h1>{{ _('Special Edition Add-on Creator T-shirt') }}</h1>
```
Comment (magopian) [inline apps/users/templates/users/t-shirt.html]: Not indented?

File: apps/users/templates/users/t-shirt.html [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+{% if eligible %}
+  {% if submitted %}
+  <section class="island c user-input prose">
```
Comment (magopian) [inline apps/users/templates/users/t-shirt.html]: Not indented?

File: apps/users/models.py (hunk 1 above)
Comment (magopian) [inline apps/users/models.py]: I see it is required by the carrier ;)

File: apps/users/views.py [hunk 4]
```diff
@@ -231,6 +232,78 @@ def edit(request):
 
 @write
 @login_required
+def t_shirt(request):
+    if not waffle.switch_is_active('t-shirt-orders'):
+        raise http.Http404()
+
+    MIN_PERSONA_ADU = 10000
+
+    user = UserProfile.objects.get(pk=request.user.id)
```
Comment (magopian) [inline apps/users/views.py]: Why not simply use `request.user`? And `request.user.tshirtorder_set.get()` for the shirt order?

File: apps/users/views.py [hunk 5]
```diff
@@ -231,6 +232,78 @@ def edit(request):
...
+            Q(addon__is_listed=False,
+              addon__versions__files__is_signed=True))
+        .exists() or
+
+        Addon.objects.filter(
+            authors=user,
+            type=amo.ADDON_PERSONA,
+            status=amo.STATUS_PUBLIC,
+            disabled_by_user=False)
+        .aggregate(Sum('average_daily_users')) > MIN_PERSONA_ADU)
```
Comment (magopian) [inline apps/users/views.py]: You're comparing a dict to an integer. In my test, it's comparing `{'average_daily_users__sum': None} > 10000` which returns True, for some odd reason. So everybody is always be eligible.

I think tests for this "eligible" test would be welcome ;) Maybe make `eligible` a helper function? That would be easier to test?

**Comment by magopian** [comment]
This looks very nice! Let me know when you fixed the `eligible` part so I can merge!

r+wc

File: apps/users/templates/users/t-shirt.html (hunk 2 above)
Comment (kmaglione) [inline apps/users/templates/users/t-shirt.html]: Well, I copied the initial version of this from another `users` template. None of the content blocks are indented from their template tags here, so I figured I should stick with the same style.

File: apps/users/views.py (hunk 4 above)
Comment (kmaglione) [inline apps/users/views.py]: Ah, sorry. I copied this view from another one that was created when we still had separate AMO and Django users.

**Pull request merged**